storage/google/cloud/storage/client.py

```diff
@@ -242,6 +242,8 @@ def create_bucket(self, bucket_name, requester_pays=None, project=None):
         If the bucket already exists, will raise
         :class:`google.cloud.exceptions.Conflict`.
 
+        See also: :meth:`~.storage.Bucket.create`
```

Comment:
I think we could be even more explicit about why someone should prefer the Bucket.create function.

Something like: To set additional properties when creating a bucket such as (link to Bucket.location), use (link to Bucket.create).

Did you think more about having this method accept a bucket or bucket name as the first argument?

Reply:
Thanks for the feedback will update documentation.

As for the `create_bucket` method, what if we were to remove this method from documentation as we shouldn't deprecate GA commands and instead surface `Bucket.create()` in this documentation? This also has a draw back this documentation is meant for `Client` and not `Bucket`..

Another approach is to update the `create_bucket(..)` to have an additional parameter `bucket` to that accepts a Bucket object with defined parameters, but I feel that if a user defines a Bucket object with required parameters they are already one step away from `Bucket.create` instead of having to call `Client.create_bucket`.

Reply:
Yeah, we shouldn't deprecate methods in a GA library, but we can encourage one method over the other. It's part of the [Python philosophy](https://www.python.org/dev/peps/pep-0020/) that "There should be one-- and preferably only one --obvious way to do it."

I think we are just disagreeing on which method should be the one obvious way.

The generated libraries have all API requests as methods of the `Client` object, and that's the direction we went for the BigQuery library too, so if we were to deprecate/discourage anything, I'd prefer we discourage `Bucket.create()` even if we end up making `create_bucket()` call it under the covers.

Since you need to provide a bucket name to create a `Bucket` object in the first place, I think it would make sense for the first parameter here to accept either a string or a Bucket.